server/modules/authentication/discord/authentication.js

```diff
@@ -19,9 +31,16 @@ module.exports = {
         passReqToCallback: true
       }, async (req, accessToken, refreshToken, profile, cb) => {
         try {
-          if (conf.guildId && !_.some(profile.guilds, { id: conf.guildId })) {
+          if (conf.roles) {
+            const discord = new DiscordOauth2()
```

Comment:
I would move the line with `discord` initialization higher, above the scope of the `passport.use` method so it won't create new instance of the DiscordOauth2 class every register 👍